### ISSUE TITLE
Method for rendering masks from instance segmentation kornia.utils.draw_masks(image, masks, color, alpha=0.5)

### ISSUE THREAD

**Issue by medphisiker** [kornia/kornia#2512]
Hello, 
thank you for the great library. 

## 🚀 Feature
I wanted to suggest adding one useful method for visualizing predictions from a neural network for instance segmentation like this ([link](https://pytorch.org/vision/stable/auto_examples/plot_visualization_utils.html#visualizing-segmentation-masks)).

In this case, the color of the masks is black, but it could be any other. And it would also be possible to specify an individual color for each mask.

![image](https://github.com/kornia/kornia/assets/4876436/3cb5bee9-1cc4-4b78-8c3f-65582cbc66c5)

It would be great to add a method for rendering masks from instance segmentation.

It would be great to make an API similar to kornia.utils.draw_rectangle(image, rectangle, color=None, fill=None)[[source]](https://kornia.readthedocs.io/en/latest/_modules/kornia/utils/draw.html#draw_rectangle)

kornia.utils.draw_masks(image, masks, color, alpha=0.5)

PARAMETERS:
image ([Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)) – is tensor of BxCxHxW.

masks ([Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)) – represents number of binary masks to draw in BxNxHxW N is the number of binary masks with same shapes as image HxW to draw per batch.

color ([Optional](https://docs.python.org/3/library/typing.html#typing.Optional)[[Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)], optional) – a size 1, size 3, BxNx1, or BxNx3 tensor. If C is 3, and color is 1 channel it will be broadcasted. Default: None

alpha[float](https://docs.python.org/3/library/functions.html#bool)], optional) – transparency of mask colors Default: 0.5

RETURN TYPE:
[Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)

RETURNS:
This operation modifies image inplace but also returns the drawn tensor for convenience with same shape the of the input BxCxHxW.

## Motivation

Visualization of masks instance segmentation by OpenCV are slow. This operation is also associated with the transformation of matrices and it seems that it will happen faster on the GPU.

## Pitch

It would be great to have a convenient method for visualizing masks on an image like ([link](https://pytorch.org/vision/stable/auto_examples/plot_visualization_utils.html#visualizing-segmentation-masks))

## Alternatives

Now there is a function from torchvision for this ([link](https://pytorch.org/vision/stable/auto_examples/plot_visualization_utils.html#visualizing-segmentation-masks)).
As far as I understand, this function uses PIL and everything works on the CPU

______________________________________________________________________

#### Consider also to contribute to Kornia universe projects :)

<sub>

- [**Tutorials**](https://github.com/kornia/tutorials): our repository containing the tutorials.

</sub>

**Comment by johnnv1** [comment]
Sure, this will be an excellent method! here is a version I have done for one tutorial, perhaps this will be a good starting point


```python
def colorize_masks(binary_masks: torch.Tensor, merge: bool = True, alpha: None | float = None) -> list[torch.Tensor]:
    """Convert binary masks (B, C, H, W), boolean tensors, into masks with colors (B, (3, 4) , H, W) - RGB or RGBA. Where C refers to the number of masks.
    Args:
        binary_masks: a batched boolean tensor (B, C, H, W)
        merge: If true, will join the batch dimension into a unique mask.
        alpha: alpha channel value. If None, will generate RGB images

    Returns:
        A list of `C` colored masks.
    """
    B, C, H, W = binary_masks.shape
    OUT_C = 4 if alpha else 3

    output_masks = []

    for idx in range(C):
        _out = torch.zeros(B, OUT_C, H, W, device=binary_masks.device, dtype=torch.float32)
        for b in range(B):
            color = torch.rand(1, 3, 1, 1, device=binary_masks.device, dtype=torch.float32)
            if alpha:
                color = torch.cat([color, torch.tensor([[[[alpha]]]], device=binary_masks.device, dtype=torch.float32)], dim=1)

            to_colorize = binary_masks[b, idx, ...].view(1, 1, H, W).repeat(1, OUT_C, 1, 1)
            _out[b, ...] = torch.where(to_colorize, color, _out[b, ...])
        output_masks.append(_out)

    if merge:
        output_masks = [c.max(dim=0)[0] for c in output_masks]

    return output_masks

```

**Comment by edgarriba** [comment]
I advocate to make `kornia_moons` our library for data viz using third-party like opencv or matplotlib
https://ducha-aiki.github.io/kornia_moons/

/cc @ducha-aiki

**Comment by edgarriba** [comment]
> Perhaps this will allow us to make the preprocessing and postprocessing of the image as part of the neural network itself. And we can compile it all together(preprocessing, deep learning model, postprocessing) as a single model in ONYX and TensorRT.

apart from differentiability, that was one of the main purposes of kornia when we expanded to a more generic cv library.

**Comment by edgarriba** [comment]
If anyone is interested in this, feel free to re-open the ticket. Thanks!
